### ISSUE TITLE
sbt assembly fails with a lot of duplicate errors

### ISSUE THREAD

**Issue by reactormonk** [CommBank/ebenezer#119]
I ran `sbt assembly`. Maybe a merge strategy is missing?

    [error] deduplicate: different file contents found in the following:
    [error] ~/.ivy2/cache/org.apache.thrift/libthrift/jars/libthrift-0.9.0-cdh5-2.jar:org/apache/thrift/transport/TSocket.class
    [error] ~/.ivy2/cache/org.apache.hive/hive-exec/jars/hive-exec-0.13.1-cdh5.2.4.jar:org/apache/thrift/transport/TSocket.class
    [error] deduplicate: different file contents found in the following:
    [error] ~/.ivy2/cache/org.apache.thrift/libthrift/jars/libthrift-0.9.0-cdh5-2.jar:org/apache/thrift/transport/TStandardFile.class
    [error] ~/.ivy2/cache/org.apache.hive/hive-exec/jars/hive-exec-0.13.1-cdh5.2.4.jar:org/apache/thrift/transport/TStandardFile.class
    [error] deduplicate: different file contents found in the following:
    [error] ~/.ivy2/cache/org.apache.thrift/libthrift/jars/libthrift-0.9.0-cdh5-2.jar:org/apache/thrift/transport/TTransport.class
    [error] ~/.ivy2/cache/org.apache.hive/hive-exec/jars/hive-exec-0.13.1-cdh5.2.4.jar:org/apache/thrift/transport/TTransport.class
    [error] deduplicate: different file contents found in the following:
    [error] ~/.ivy2/cache/org.apache.thrift/libthrift/jars/libthrift-0.9.0-cdh5-2.jar:org/apache/thrift/transport/TTransportException.class
    [error] ~/.ivy2/cache/org.apache.hive/hive-exec/jars/hive-exec-0.13.1-cdh5.2.4.jar:org/apache/thrift/transport/TTransportException.class
    [error] deduplicate: different file contents found in the following:
    [error] ~/.ivy2/cache/org.apache.thrift/libthrift/jars/libthrift-0.9.0-cdh5-2.jar:org/apache/thrift/transport/TTransportFactory.class
    [error] ~/.ivy2/cache/org.apache.hive/hive-exec/jars/hive-exec-0.13.1-cdh5.2.4.jar:org/apache/thrift/transport/TTransportFactory.class
    [error] deduplicate: different file contents found in the following:
    [error] ~/.ivy2/cache/org.datanucleus/datanucleus-rdbms/jars/datanucleus-rdbms-3.2.9.jar:plugin.xml
    [error] ~/.ivy2/cache/org.datanucleus/datanucleus-api-jdo/jars/datanucleus-api-jdo-3.2.6.jar:plugin.xml
    [error] ~/.ivy2/cache/org.datanucleus/datanucleus-core/jars/datanucleus-core-3.2.10.jar:plugin.xml

**Comment by vineethvarghese** [comment]
@reactormonk from the root of the project you need to run `sbt package` to get library jars. These are then used in your project where you can use `sbt assembly` to create the fat jar. Please take a look at the example project which uses the ebenezer libraries. You can run `sbt assembly` from there.

You can also take a look at the travis logs https://travis-ci.org/CommBank/ebenezer

Let us know if you still have issues.

**Comment by reactormonk** [comment]
Nope, `sbt package` doesn't help. I called `sbt assembly` from the root directory of this project for https://github.com/CommBank/ebenezer#introspection-cli

**Comment by vineethvarghese** [comment]
Sorry about the confusion but the documentation is not upto date. I gather you want to run the CLI tool. Once you start sbt from root, your will need to get into the tools project via running `project tools`. Here you can run `sbt assembly` and get the ebenzer-tool-assembly-*.jar which will have the CLI bundled inside.

What exactly are you trying to do? I gather you are trying to read a parquet file and just want to use the CLI for this?

**Comment by reactormonk** [comment]
In the end, I want to read a parquet file from spark, but I wanted to start with some introspection to see what the thrift files should look like.

**Comment by PCejrowski** [comment]
Hi, have you managed to solved the problem ?

**Comment by reactormonk** [comment]
@PCejrowski I went with creating a SparkContext to store and load the data into hdfs. E.g.

    sqlContext.createDataFrame(data).write.parquet(s"/input")

**Comment by PCejrowski** [comment]
@reactormonk Ok, thank you, 
I have just managed to solve similar issue using `exclude` and MergeStrategy combination.
